Rakefile

```diff
@@ -29,3 +36,87 @@ RSpec::Core::RakeTask.new(:spec) do |r|
 end
 
 task :default => :spec
+
+
+class EmptyMiddleware
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    if @app.respond_to?(:call)
+      status, headers, body = @app.call(env)
+    else
+      [200, {}, 'Hello world']
+    end
+  end
+end
+
+class FaradayMiddleware
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    conn = Faraday.new(url: 'http://www.google.com') do |faraday|
+      faraday.use ZipkinTracer::FaradayHandler
+      # default Faraday stack
+      faraday.request :url_encoded
+      faraday.adapter Faraday.default_adapter
+    end
+   # conn.get('/')
+    [200, {}, 'hello world']
+  end
+end
+
+task :run_once do
+  empty_app = EmptyMiddleware.new(nil)
+  app = FaradayMiddleware.new(nil)
+
+  logger = Logger.new(Tempfile.new('fakelog'))
+  null_configuration = { logger: logger, sample_rate: 1}
+  null_tracer_rack = ZipkinTracer::RackHandler.new(app, null_configuration)
+  env = Rack::MockRequest.env_for('/path', {})
+
+  sleep(10)
+  null_tracer_rack.call(env)
+end
+
+# This task is used to help development of ZipkinTracer.
+desc "Runs benchmarks for ZipkinTracer."
+task :benchmark do
+  logger = Logger.new(Tempfile.new('fakelog'))
+  fake_url = 'http://www.google.com' #resolve but unable to send I hope!
+
+  empty_app = EmptyMiddleware.new(nil)
+  app = FaradayMiddleware.new(nil)
+
+  null_configuration = { logger: logger, sample_rate: 1}
+  json_configuration = null_configuration.merge({ json_api_host: fake_url })
+  scribe_configuration = null_configuration.merge({ scribe_server: fake_url })
+
+  empty_rack = EmptyMiddleware.new(empty_app)
+  null_tracer_rack = ZipkinTracer::RackHandler.new(empty_app, null_configuration)
+  null_tracer_faraday_rack = ZipkinTracer::RackHandler.new(app, null_configuration)
+  json_tracer_rack = ZipkinTracer::RackHandler.new(empty_app, json_configuration)
+  json_tracer_faraday_rack = ZipkinTracer::RackHandler.new(app, json_configuration)
+  scribe_tracer_rack = ZipkinTracer::RackHandler.new(empty_app, scribe_configuration)
+  scribe_tracer_faraday_rack = ZipkinTracer::RackHandler.new(app, scribe_configuration)
+
+  env = Rack::MockRequest.env_for('/path', {})
+
+  Benchmark.ips do |bm|
+ #   bm.report("No rack middleware") { empty_rack.call(env) } # Uncomment if curious
```

Comment:
I think it is useful for someone trying to know the performance hit in general. Comparing with a rack middleware doing nothign allows that. I do not leave it uncommented because usually everything looks so slow in comparison and it is difficult to compare zipkin implementations